spring-cloud-contract-stub-runner/src/test/groovy/org/springframework/cloud/contract/stubrunner/AetherStubDownloaderSpec.groovy

```diff
@@ -27,4 +29,28 @@ class AetherStubDownloaderSpec extends Specification {
         then:
         jar != null
     }
+
+    def 'Should use local repository from settings.xml'() {
```

Comment:
AFAIR you can annotate this test with `@RestoreSystemProperties` - that way whatever props you set in the test will get restored afterwards.